src/processor/operator/partitioner.cpp

```diff
@@ -46,11 +46,13 @@ void PartitionerSharedState::initialize(const PartitionerDataInfo& dataInfo) {
     Partitioner::initializePartitioningStates(dataInfo, partitioningBuffers, numPartitions);
 }
 
-partition_idx_t PartitionerSharedState::getNextPartition(idx_t partitioningIdx) {
+partition_idx_t PartitionerSharedState::getNextPartition(idx_t partitioningIdx,
+    std::shared_ptr<RelBatchInsertProgressSharedState> progressSharedState) {
```

Comment:
```suggestion
    RelBatchInsertProgressSharedState& progressSharedState) {
```
pass in reference should be enough here, as you don't actually share the ownership.